llvm/include/llvm/ADT/APFloat.h

```diff
@@ -1536,6 +1536,26 @@ inline APFloat abs(APFloat X) {
   return X;
 }
 
+/// Returns X^N for N >= 0.
+inline APFloat powi(const APFloat &X, int64_t N) {
+  assert(N >= 0 && "negative exponents not supported.");
+  APFloat Acc = APFloat::getOne(X.getSemantics());
+  if (N == 0) {
+    return Acc;
```

Comment:
This should canonicalize the value

Reply:
For N==0? Can you explain?

Reply:
All paths through the function, but yes this one too. Signaling nans should be quieted